build_tools/cmake/iree_run_module_test.cmake

```
@@ -0,0 +1,276 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Utility function to check expected failure for different platforms.
+function(platform_xfail OUTPUT_XFAIL XFAIL_ARGS)
+  set(${OUTPUT_XFAIL} FALSE PARENT_SCOPE)
+  if(XFAIL_ARGS)
+    if((CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+        CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+        XFAIL_ARGS MATCHES "rv64") OR
+       (CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv32" AND
+        CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+        XFAIL_ARGS MATCHES "rv32") OR
+       (ANDROID AND XFAIL_ARGS MATCHES "arm64-v8a") OR
+       (XFAIL_ARGS MATCHES "x86_64"))
```

We can use on list operators rather than relying on CMake's weird stringly-typed list. Also note that the current approach will fail if any platform is a substring of another.

```suggestion
        XFAIL_ARGS MATCHES "rv64") OR
       (CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv32" AND
        CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
        "rv32" IN_LIST XFAIL_ARGS) OR
       (ANDROID AND "arm64-v8a" IN_LIST XFAIL_ARGS) OR
       ("x86_64" IN_LIST XFAIL_ARGS))
```